File: exercises/weave/queue.js
// --- Directions
// Implement a 'peek' method in this Queue class.
// Peek should return the first element (the next
// one to be dequeued) from the queue *without*
// removing it.

class Queue {
  constructor () {
    this.data = [];
  }

  add (record) {
    this.data.unshift(record);
  }

  remove () {
    return this.data.pop();
  }

  peek () {
    // return the first element in the queue
    // (return the one to be dequeued next w/o actually dequeuing it)
    return this.data[this.data.length - 1];
  }
}

module.exports = Queue;

File: exercises/weave/queue.py
#!/usr/bin/env python3


class Queue:

    def __init__(self):
        self.data = []

    def enqueue(self, elem):
        self.data.insert(0, elem)

    def dequeue(self):
        return self.data.pop()

    def peek(self):
        try:
            return self.data[-1]
        except:
            return None
